File: frontend/src/types/transaction.ts
export interface Transaction {
  id: number;
  description: string;
  amount: number;
  type: 'income' | 'expense';
  date: string;
  category_id?: number;
  category_name?: string;
  tags?: string;
  notes?: string;
  is_recurring: boolean;
  is_verified: boolean;
  confidence_score?: number;
  invoice_id?: number;
  created_at: string;
  updated_at: string;
}

export interface TransactionCreate {
  description: string;
  amount: number;
  type: 'income' | 'expense';
  date: string;
  category_id?: number;
  tags?: string;
  notes?: string;
}

export interface TransactionUpdate {
  description?: string;
  amount?: number;
  type?: 'income' | 'expense';
  date?: string;
  category_id?: number;
  tags?: string;
  notes?: string;
  is_verified?: boolean;
}

File: backend/app/schemas/transaction.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional

class TransactionBase(BaseModel):
    description: str
    amount: float
    type: str = Field(..., pattern="^(income|expense)$")
    date: datetime
    tags: Optional[str] = None
    notes: Optional[str] = None

class TransactionCreate(TransactionBase):
    category_id: Optional[int] = None

class TransactionUpdate(BaseModel):
    description: Optional[str] = None
    amount: Optional[float] = None
    type: Optional[str] = None
    date: Optional[datetime] = None
    category_id: Optional[int] = None
    tags: Optional[str] = None
    notes: Optional[str] = None
    is_verified: Optional[bool] = None

class TransactionResponse(TransactionBase):
    id: int
    category_id: Optional[int]
    category_name: Optional[str] = None
    is_recurring: bool
    is_verified: bool
    confidence_score: Optional[float]
    invoice_id: Optional[int]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True